lib/Sso/Session.php

```diff
@@ -135,6 +135,14 @@ public function performRemoteCheck($httpClient = null) {
       return false;
     }
 
+    /**
+     * @return the Idp logout url where users should be redirected to upon logging out
+     */
+    public function getLogoutUrl()
+    {
+      return Maestrano::with($this->_preset)->sso()->getLogoutUrlWithUserUid($this->uid);
```

Comment:
would it be possible to assign Maestrano::with($this->_preset) to a variable in the whole file ? ( I suppose more refactoring is needed...)